src/store/subscriber_store.rs

```diff
@@ -0,0 +1,88 @@
+use std::collections::HashMap;
+
+use crate::model::{NewSubscriber, Subscriber};
+
+pub trait SubscriberStore {
+    async fn create(&mut self, new_subscriber: NewSubscriber) -> Subscriber;
+    async fn all(&self) -> Vec<Subscriber>;
+    async fn delete(&mut self, id: i32) -> ();
+}
+
+#[derive(Debug, Default)]
+pub struct InMemorySubscriberStore {
+    subscribers: HashMap<i32, Subscriber>,
+    next_id: i32,
+}
+
+impl SubscriberStore for InMemorySubscriberStore {
+    async fn create(&mut self, new_subscriber: NewSubscriber) -> Subscriber {
+        let id = self.get_next_id();
+        let subscriber = Subscriber {
+            id,
+            email: new_subscriber.email,
```

Comment:
Do we want to have the behaviour of only letting an email be subscribed once? I feel like that'd be nice?

Reply:
Probably!